aix,ibmi: _PATH_DEFPATH and NAME_MAX undefined after posix_spawn changes

On aix, ibmi after this [commit](https://github.com/libuv/libuv/commit/40d45efebf9bf1a8091a573d26e53d9c4262ef5b#diff-b8514019621b0d35d95e9f96451ae5a07de9afd56a5b15c72a910de78e8e8891), we are getting this failure during build on upstream [libuv ci](https://ci.nodejs.org/job/libuv-test-commit-ibmi/nodes=ibmi74-ppc64/2092/console) 

Error is 
```c
src/unix/process.c: In function 'uv__spawn_resolve_and_spawn':
src/unix/process.c:764:12: error: '_PATH_DEFPATH' undeclared (first use in this function)
  764 |     path = _PATH_DEFPATH;
      |            ^~~~~~~~~~~~~
src/unix/process.c:764:12: note: each undeclared identifier is reported only once for each function it appears in
src/unix/process.c:766:30: error: 'NAME_MAX' undeclared (first use in this function)
  766 |   k = strnlen(options->file, NAME_MAX + 1);
      |                              ^~~~~~~~
```

In process.c, [code](https://github.com/libuv/libuv/commit/83efa3dd7140ef16e9d8fcde42011450f0c229e6) worked previously because it was for only macOS. [Commit](https://github.com/libuv/libuv/commit/40d45efebf9bf1a8091a573d26e53d9c4262ef5b#diff-b8514019621b0d35d95e9f96451ae5a07de9afd56a5b15c72a910de78e8e8891) removed the #if defined(__APPLE__) conditionals and made posix_spawn the default path for all Unix systems. However, the code now unconditionally includes <paths.h> and uses _PATH_DEFPATH and NAME_MAX

But since we don't have NAME_MAX defined in limits.h/unistd.h and _PATH_DEFPATH defined in paths.h we are facing error on aix/ibmi.

--- a/src/unix/process.c
+++ b/src/unix/process.c
@@ -37,6 +37,15 @@
 #include <paths.h>
 #include <dlfcn.h>
 
+#if defined(__PASE__)
+#define _PATH_DEFPATH "/QOpenSys/pkgs/bin:/usr/bin:/bin"
+#elif defined(_AIX)
+#define _PATH_DEFPATH "/opt/freeware/bin:/usr/bin:/bin"
+#endif
+#ifndef NAME_MAX
+#define NAME_MAX 255
+#endif
+
 #if defined(__APPLE__)
 # include <sys/kauth.h>
 # include <sys/sysctl.h>
@@ -455,7 +464,7 @@ static void uv__spawn_init_can_use_setsid(void) {
 
 
 static void uv__spawn_init_posix_spawn(void) {
-#if !defined(__linux__)
+#if !defined(__linux__) && !defined(_AIX) && !defined(__PASE__)
   posix_spawn_works = 1;
 #elif !defined(__ANDROID__)
   pid_t pid;

--- a/src/unix/process.c
+++ b/src/unix/process.c
@@ -38,7 +38,7 @@
 #include <dlfcn.h>
 
 #if defined(__PASE__)
-#define _PATH_DEFPATH "/QOpenSys/pkgs/bin:/usr/bin:/bin"
+#define _PATH_DEFPATH "/QOpenSys/pkgs/bin:/QOpenSys/usr/bin:/usr/bin"
 #elif defined(_AIX)
 #define _PATH_DEFPATH "/opt/freeware/bin:/usr/bin:/bin"
 #endif

